internal/pkg/skuba/util/slices.go

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package util
+
+func UniqueStringSlice(slice []string) []string {
+	result := []string{}
+	seen := make(map[string]struct{}, len(slice))
+	for _, value := range slice {
+		if _, ok := seen[value]; !ok {
+			seen[value] = struct{}{}
```

Comment:
Can you explain a bit what this `{}{}` is supposed to mean? Looks like an arcane Perl spell script to me :P

Reply:
I assume it is for memory optimization - it seems a empty `struct{}` consumes no memory, whereas a `bool` would consume 1 byte already to track if a string was already seen before.

Reply:
Yes, exactly, in this case we are only interested in `seen` values, that are keys in the map, the values of those keys are irrelevant, that's why an empty struct pays the bill. `struct{}{}` is just another way of saying `make(struct{})`.